app/helpers/tournaments_helper.rb

```diff
@@ -1,2 +1,34 @@
 module TournamentsHelper
+
+    def success_per_question(results)
+        statistics = Array.new;
+        questions = Array.new;
+    
+        # initialize empty array
+        for index in 0..9 do
+          questions[index] = 0;
+        end
+    
+        results.each do |result|
+          # count all true questions
+          for index in 1..10 do
+            if (result.answers[index.to_s])
+              questions[index-1] = questions[index-1] + 1
+            end
+          end
+        end
+    
```

Comment:
```suggestion
        results.each {|result| countCorrectAnswersInResult(result.answers, questions}
        
def countCorrectAnswersInResult(resultAnswers, questions)
	for index in 1..10 do
        if (resultAnswers[index.to_s])
            questions[index-1] += 1
        end
     end
end
```

This can probably be simplified even more, also can't say I'm a huge fan of mutating the `questions` array like that, but up to you if you want to use this snippet or think of some other way of doing it. The idea is to move the internal loop to another function